osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisServiceImpl.java

```diff
@@ -484,34 +484,50 @@ public OsisS3Capabilities getS3Capabilities() {
+            }
 
-                Credentials tempCredentials = getCredentials(tenantId);
-                final AmazonIdentityManagement iam = vaultAdmin.getIAMClient(tempCredentials,
-                        appEnv.getRegionInfo().get(0));
+            if (tenantId == null || tenantId.isEmpty() || userId == null || userId.isEmpty()) {
+                logger.info("Delete S3 Credential: Missing tenantId/userId in request, call Vault superAdmin API GetUserByAccessKey");
+                Map<String, String> result = getTenantIdAndUserIdByAccessKeyFromVault(accessKey);
+                tenantId = result.get("tenantId");
+                userId = result.get("userId");
+                logger.info("Delete S3 Credential: GetUserByAccessKey response received, tenant id: {}, user id: {}", tenantId, userId);
+            }
 
-                DeleteAccessKeyRequest deleteAccessKeyRequest = ScalityModelConverter
-                        .toDeleteAccessKeyRequest(accessKey, userId);
+            Credentials tempCredentials = getCredentials(tenantId);
+            final AmazonIdentityManagement iam = vaultAdmin.getIAMClient(tempCredentials,
+                    appEnv.getRegionInfo().get(0));
 
-                logger.debug("[Vault] Delete Access Key Request:{}", new Gson().toJson(deleteAccessKeyRequest));
+            DeleteAccessKeyRequest deleteAccessKeyRequest = ScalityModelConverter
+                    .toDeleteAccessKeyRequest(accessKey, userId);
 
-                DeleteAccessKeyResult deleteAccessKeyResult = iam.deleteAccessKey(deleteAccessKeyRequest);
+            logger.debug("[Vault] Delete Access Key Request:{}", new Gson().toJson(deleteAccessKeyRequest));
 
-                logger.debug("[Vault] Delete Access Key response:{}", new Gson().toJson(deleteAccessKeyResult));
+            DeleteAccessKeyResult deleteAccessKeyResult = iam.deleteAccessKey(deleteAccessKeyRequest);
 
-                deleteSecretKey(ScalityModelConverter.toRepoKeyForCredentials(userId, accessKey));
+            logger.debug("[Vault] Delete Access Key response:{}", new Gson().toJson(deleteAccessKeyResult));
 
-                logger.info("Delete S3 credential successful:: tenant ID:{}, user ID:{}, accessKey:{}",
-                        tenantId, userId, accessKey);
+            deleteSecretKey(ScalityModelConverter.toRepoKeyForCredentials(userId, accessKey));
 
-            } catch (Exception e) {
-                logger.error("Delete S3 credential failed. Error details:", e);
+            logger.info("Delete S3 credential successful:: tenant ID:{}, user ID:{}, accessKey:{}",
+                    tenantId, userId, accessKey);
+
+        } catch (Exception e) {
+            if (isAdminPolicyError(e)) {
+                try {
+                    generateAdminPolicy(tenantId);
+                    deleteS3Credential(tenantId, userId, accessKey);
+                } catch (Exception ex) {
```

Comment:
Nit: the nested try/catch in the PRs could be avoided to simplify the code (but I agree, this is not the highest priority today :smile: ).